aim/storage/query.py

```diff
@@ -116,13 +106,18 @@ def __call__(
 
 @lru_cache(maxsize=100)
 def compile_checker(expr):
-    source_code = CODE_FORMAT.format(expr=expr)
+    source_code = expr
     byte_code = compile_restricted(source_code,
                                    filename='<inline code>',
-                                   mode='exec')
-    namespace = dict()
-    exec(byte_code, restricted_globals, namespace)
-    return namespace['check']
+                                   mode='eval')
+    return byte_code
+
+class LazyLocals(dict):
+    def __getitem__(self, key: str):
+        try:
+            return super().__getitem__(key)
+        except KeyError:
+            return self['fallback'][key]
```

Comment:
should we handle the case when there's no 'fallback' key? I do understand that this is used in the scope of this file, so feel free to skip.

Reply:
If there is no fallback object to look for the key, we might want to raise KeyError (as our locals do not possess the key).
When we return self['fallback'][key] it will automatically raises KeyError as there is no 'fallback' in LazyLocals.

Reply:
got it, thanks!